widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java

```diff
@@ -184,41 +186,58 @@ private boolean isPermissionDataMediaRequestCode(Pair<Integer, Integer[]> permis
     }
 
     private void onCallPermissionsAvailable() {
+        GliaSdkConfiguration sdkConfiguration = configuration.getSdkConfiguration();
         callView.startCall(
-                configuration.getCompanyName(),
-                configuration.getQueueId(),
-                configuration.getContextUrl(),
-                configuration.getUseOverlay(),
-                configuration.getScreenSharingMode(),
-                getMediaType()
+                sdkConfiguration.getCompanyName(),
+                sdkConfiguration.getQueueId(),
+                sdkConfiguration.getContextUrl(),
+                sdkConfiguration.getUseOverlay(),
+                sdkConfiguration.getScreenSharingMode(),
+                configuration.getMediaType()
         );
     }
 
-    private Engagement.MediaType getMediaType() {
-        String mediaType = getIntent().getStringExtra(GliaWidgets.MEDIA_TYPE);
-        if (mediaType != null && mediaType.equals(GliaWidgets.MEDIA_TYPE_VIDEO)) {
-            return Engagement.MediaType.VIDEO;
-        } else {
-            return Engagement.MediaType.AUDIO;
-        }
-    }
-
     private void navigateToChat() {
         Logger.d(TAG, "navigateToChat");
+        GliaSdkConfiguration sdkConfiguration = configuration.getSdkConfiguration();
         Intent newIntent = new Intent(getApplicationContext(), ChatActivity.class);
-        newIntent.putExtra(GliaWidgets.COMPANY_NAME, configuration.getCompanyName());
-        newIntent.putExtra(GliaWidgets.QUEUE_ID, configuration.getQueueId());
-        newIntent.putExtra(GliaWidgets.CONTEXT_URL, configuration.getContextUrl());
-        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.getRunTimeTheme());
-        newIntent.putExtra(GliaWidgets.USE_OVERLAY, configuration.getUseOverlay());
-        newIntent.putExtra(GliaWidgets.SCREEN_SHARING_MODE, configuration.getScreenSharingMode());
+        newIntent.putExtra(GliaWidgets.COMPANY_NAME, sdkConfiguration.getCompanyName());
+        newIntent.putExtra(GliaWidgets.QUEUE_ID, sdkConfiguration.getQueueId());
+        newIntent.putExtra(GliaWidgets.CONTEXT_URL, sdkConfiguration.getContextUrl());
+        newIntent.putExtra(GliaWidgets.UI_THEME, sdkConfiguration.getRunTimeTheme());
+        newIntent.putExtra(GliaWidgets.USE_OVERLAY, sdkConfiguration.getUseOverlay());
+        newIntent.putExtra(GliaWidgets.SCREEN_SHARING_MODE, sdkConfiguration.getScreenSharingMode());
         startActivity(newIntent);
     }
 
     private void navigateToSurvey(Survey survey) {
         Intent newIntent = new Intent(getApplicationContext(), SurveyActivity.class);
-        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.getRunTimeTheme());
+        newIntent.putExtra(GliaWidgets.UI_THEME, configuration.getSdkConfiguration().getRunTimeTheme());
         newIntent.putExtra(GliaWidgets.SURVEY, (Parcelable) survey);
         startActivity(newIntent);
     }
+
+    @Deprecated
```

Comment:
@deniss-mc Suggest adding a comment explaining what is the recommended alternative for this deprecated function.

Reply:
Added comments, please review if needed @andrews-moc